src/System.Device.Gpio/System/Device/Gpio/GpioPin.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Represents a general-purpose I/O (GPIO) pin.
+    /// </summary>
+    public sealed class Gpio​Pin : IDisposable
+    {
+        private readonly int _pinNumber;
+        private readonly GpioController _controller;
+        private bool _disposedValue;
```

Comment:
if you null `controller` you can just null check rather than having this field

Reply:
the controller is not nulled as not disposed at all.